x/batch/decorator.go

```diff
@@ -0,0 +1,167 @@
+/*
+Package batch provides batch transaction support
+middleware to support multiple operations in one
+transaction
+*/
+package batch
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/common"
+)
+
+const MaxBatchMessages = 10
+
+//----------------- Decorator ----------------
+//
+// This is just a binding from the functionality into the
+// Application stack, not much business logic here.
+
+// Decorator iterates through batch transaction messages and passes them down the stack
+type Decorator struct {
+}
+
+var _ weave.Decorator = Decorator{}
+
+// NewDecorator returns a batch transaction decorator
+func NewDecorator() Decorator {
+	return Decorator{}
+}
+
+type BatchTx struct {
+	weave.Tx
+	Msg weave.Msg
+}
+
+func (tx *BatchTx) GetMsg() (weave.Msg, error) {
+	return tx.Msg, nil
+}
+
+// Check iterates through messages in a batch transaction and passes them
+// down the stack
+func (d Decorator) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx,
+	next weave.Checker) (weave.CheckResult, error) {
+	var res weave.CheckResult
+	var err error
+	msg, err := tx.GetMsg()
+	if err != nil {
+		return res, err
+	}
+
+	if batchMsg, ok := msg.(Msg); ok {
```

Comment:
pp: if you would invert the condition, you can have the happy path aligned to the left.